ibis/selectors.py

```diff
@@ -366,7 +367,16 @@ def all_of(*predicates: str | Predicate) -> Predicate:
 def c(*names: str) -> Predicate:
     """Select specific column names."""
     names = frozenset(names)
-    return where(lambda col: col.get_name() in names)
+
+    def func(col: ir.Value) -> bool:
+        schema = col.op().table.schema
+        if extra_cols := (names - schema.keys()):
+            raise exc.IbisInputError(
+                f"Columns {extra_cols} are not present in {schema.names}"
+            )
+        return col.get_name() in names
```

Comment:
This might be cleaner/more efficient to implement `c` directly as a subclass of `Selector`? This way you could define `expand` directly, which would give direct access to the input table rather than having to hop through `.op().table`. The column names existing check would also only happen once rather than `len(table.columns)` times.

Reply:
That is a much better idea than what I am currently doing, thank you.

Reply:
Hm, this approach turns out to break composition of selectors with `~`, `&` and `|` :(

Reply:
@jcrist Not sure if you have any other ideas here, but I'm inclined to merge this as it is.

Reply:
That's unfortunate. I think this should be doable as I suggested above, but it would require refactoring how the composition operators work. Going with what you have seems reasonable for now.